src/tcp/_Private/DisposableTCPSocket.php

```diff
@@ -0,0 +1,28 @@
+<?hh // strict
+/*
+ *  Copyright (c) 2004-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the MIT license found in the
+ *  LICENSE file in the root directory of this source tree.
+ *
+ */
+
+namespace HH\Lib\_Private;
+
+use namespace HH\Lib\Experimental\{IO, TCP};
+
+final class DisposableTCPSocket
```

Comment:
Random thought: What if we only had one class for each handle type, that implements both `__dispose()` and `close()`? We'd still have both Disposable/NonDisposable interfaces (implemented by the same class but that's an implementation detail that no one should see because the classes are `_Private`).

Reply:
Disposables can not extend non-disposable classes or use non-disposable traits. Composition is the only way to share implementation.